### DISCUSSION TITLE
No longer understand how to pass database_id to iterator since the refactor

### DISCUSSION THREAD

**Discussion by JPOvenden** [jheddings/notional#110] [Q&A] [answered]
Notional Version: 0.7.0

Using a lib that depends on this, but is now two months old, I receive an unexpected argument error when calling the iterator:

`EndpointIterator.__init__() got an unexpected keyword argument `'database_id'``

I know there have been many changes to Notional recently, I want to know what replaces this deprecated argument?

**Comment by jheddings** [comment, marked as answer]
Hi JPOvenden.  Thanks for reaching out and sorry for the trouble.  I also just noticed that the site docs were not updated with the new release, so I'll try to fix that.

The iterator no longer accepts API parameters during init.  Instead, simply pass the endpoint you want to iterate over.  When you are ready to grab the results, call the iterator using the desired API parameters.

See the [iterator.py](https://github.com/jheddings/notional/blob/main/examples/iterator.py) example for more details.

This interface should be stable, so hopefully you won't run into many more issues!

**Comment by JPOvenden** [comment]
Thank you, and thanks for your work!
